src/Form/QueueContentsForm.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Drupal\media_mpx\Form;
+
+use Drupal\Core\Batch\BatchBuilder;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\media_mpx\Repository\MpxMediaType;
+use Drupal\media_mpx\Service\QueueVideoImportsRequest;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class QueueContentsForm.
+ */
+class QueueContentsForm extends FormBase {
+
+  /**
+   * The mpx Media Types repository.
+   *
+   * @var \Drupal\media_mpx\Repository\MpxMediaType
+   */
+  private $mpxMediaTypeRepository;
+
+  /**
+   * QueueContentsForm constructor.
+   */
+  public function __construct(MpxMediaType $mpxMediaTypeRepository) {
+    $this->mpxMediaTypeRepository = $mpxMediaTypeRepository;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $mpx_media_types = $this->mpxMediaTypeRepository->findAllTypes();
+    $options = [];
+    foreach ($mpx_media_types as $mpx_media_type) {
+      $options[$mpx_media_type->id()] = $mpx_media_type->label();
+    }
+
+    $form['mpx_video_types'] = [
+      '#type' => 'checkboxes',
+      '#title' => t('mpx Video Types'),
+      '#options' => $options,
+      '#required' => TRUE,
+    ];
+    $form['queue'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Queue selected Video Types'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $types = array_filter(array_values($form_state->getValue('mpx_video_types')));
+
+    $batch_builder = new BatchBuilder();
+    $batch_builder->setTitle(t('MPX Video Imports Queueing'))
+      ->setFinishCallback([$this, 'batchFinished'])
+      ->setInitMessage(t('Starting Queueing Process.'));
+
+    foreach ($types as $type) {
+      $batch_builder->addOperation([$this, 'batchedQueueItemsForMediaType'], ['media_type_id' => $type]);
+    }
+    batch_set($batch_builder->toArray());
+  }
+
+  /**
+   * Operation callback to queue each mpx media type for imports.
+   *
+   * @param string $media_type_id
+   *   The media type id to process in this callback.
+   * @param array $context
+   *   Context data passed between batches.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\media_mpx\Exception\MediaTypeDoesNotExistException
+   * @throws \Drupal\media_mpx\Exception\MediaTypeNotAssociatedWithMpxException
+   */
+  public function batchedQueueItemsForMediaType(string $media_type_id, array &$context) {
+    /* @var \Drupal\media_mpx\Service\QueueVideoImports $service */
+    $service = \Drupal::getContainer()->get('media_mpx.service.queue_video_imports');
+    $batch_size = 200;
```

Comment:
We should document how you came up with this number.

Reply:
Added:

```
    // Mpx responses might be a bit heavy, so limit batches size to 200. Don't
    // want php running out of memory.
    // @see vendor/lullabot/mpx-php/src/DataService/ObjectListQuery.php
```